src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java

```diff
@@ -60,51 +67,91 @@ private void sendNotificationPerSubscription(NotificationWrapper notificationWra
         for (SubscriptionWithType subscriptionWithType : notificationWrapper.getSubscriptionsBasedOnNotificationType()) {
 
             if (isValidNotification(notificationWrapper, subscriptionWithType.getSubscription(), notificationType)) {
-
-                Notification notification = notificationFactory.create(notificationWrapper, subscriptionWithType.getSubscriptionType());
-
-                sendNotificationService.sendEmailSmsLetterNotification(notificationWrapper, subscriptionWithType.getSubscription(), notification);
-                processOldSubscriptionNotifications(notificationWrapper, notification);
+                sendNotification(notificationWrapper, subscriptionWithType);
                 reminderService.createReminders(notificationWrapper);
+                resendLastNotification(notificationWrapper, subscriptionWithType);
             }
         }
     }
 
+    private void resendLastNotification(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        if (shouldProcessLastNotification(notificationWrapper, subscriptionWithType)) {
+            log.info("resending the last notification");
+            scrubMobileAndSmsIfSubscribedBefore(notificationWrapper, subscriptionWithType);
+            notificationWrapper.getSscsCaseDataWrapper().setNotificationEventType(
+                    getNotificationByCcdEvent(notificationWrapper.getNewSscsCaseData().getEvents().get(0)
+                            .getValue().getEventType()));
+            sendNotification(notificationWrapper, subscriptionWithType);
+            notificationWrapper.getSscsCaseDataWrapper().setNotificationEventType(SUBSCRIPTION_UPDATED_NOTIFICATION);
+        }
+    }
+
+    private void scrubMobileAndSmsIfSubscribedBefore(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        Subscription oldSubscription = getSubscription(notificationWrapper.getOldSscsCaseData(), subscriptionWithType.getSubscriptionType());
+        Subscription newSubscription = subscriptionWithType.getSubscription();
+        String email = oldSubscription.isEmailSubscribed() ? null : newSubscription.getEmail();
+        String mobile = oldSubscription.isSmsSubscribed() ? null : newSubscription.getMobile();
+        subscriptionWithType.setSubscription(newSubscription.toBuilder().email(email).mobile(mobile).build());
+    }
+
+    private boolean shouldProcessLastNotification(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        return SUBSCRIPTION_UPDATED_NOTIFICATION.equals(notificationWrapper.getSscsCaseDataWrapper().getNotificationEventType())
+                && !isPaperCase(notificationWrapper.getNewSscsCaseData().getAppeal().getHearingType())
```

Comment:
Where is the requirement to not send the last notification if it's a paper case? Can't see that on the ticket so just want to make sure this is intended...

Reply:
I copied it from existing code. The existing code had a lot of bugs. Looking at the acceptance criteria https://tools.hmcts.net/jira/browse/SSCS-3955. This doesn't have the paper case requirement so I'll remove it.